File: www/api/app/Models/ListOption.php
<?php

namespace App\Models;

use App\Models\Traits\Attributes\ListOptionAttributes;
use App\Models\Traits\EntityPhpDoc;
use Carbon\Carbon;
use Illuminate\Database\Eloquent\Model;

/**
 * @property-read int $id
 * @property string $group_slug
 * @property string $title
 * @property int $weight
 * @property boolean $is_hidden_from_user_filters
 * @property Carbon|null $created_at
 * @property Carbon|null $updated_at
 */
class ListOption extends Model
{
    use EntityPhpDoc;
    use ListOptionAttributes;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'group_slug',
        'title',
        'weight',
        'is_hidden_from_user_filters'
    ];

    public function getMorphClass(): string
    {
        return 'ListOption';
    }
}

File: www/site-api/models/list_option.go
package models

import (
	"time"
)

// ListOption представляет опцию списка (размеры, цвета, бренды и т.д.)
type ListOption struct {
	ID                      uint      `json:"id" gorm:"primaryKey"`
	GroupSlug               string    `json:"group_slug" gorm:"type:varchar(255);not null"`
	Title                   string    `json:"title" gorm:"type:varchar(255);not null"`
	Weight                  int       `json:"-" gorm:"default:0"`
	IsHiddenFromUserFilters bool      `json:"is_hidden_from_user_filters" gorm:"default:false"`
	CreatedAt               time.Time `json:"created_at"`
	UpdatedAt               time.Time `json:"updated_at"`
}

// TableName возвращает имя таблицы для GORM
func (ListOption) TableName() string {
	return "list_options"
}
